### PR TITLE
length bugfix

### PR DESCRIPTION
**Pull request by matteofigus**
1) writing to global length can break things later (in fact it breaks Google Recaptcha)
2) returning the gpid as soon as we have it to avoid circling around the full parts.length array

Note: I am not sure the checked-in bundled files are up-to-date. Let's keep it in a separate PR?

### CODE REVIEW

File: lib/user.js [hunk 1]
```diff
@@ -88,19 +88,19 @@ User.prototype.opentable_gid = function(){
   var all_cookie_keys =  document.cookie.split("; ");
   var cookie_key = this.mobileweb_or_web_key();
   var cookie_subkey = this.mobileweb_or_web_subkey();
-  var gid = "0";
   var len = all_cookie_keys.length;
   for (i = 0; i < len; i++) {
     if (all_cookie_keys[i].match(cookie_key)) {
       var parts = all_cookie_keys[i].split("&");
-      for (x = 0, length = parts.length; x < length; x++) {
+      for (x = 0; x < parts.length; x++) {
```
Comment (pennello) [inline lib/user.js]: Use `var x`; otherwise it's in the global scope.

File: lib/user.js [hunk 2]
```diff
@@ -88,19 +88,19 @@ User.prototype.opentable_gid = function(){
...
         if (parts[x].match(cookie_subkey)) {
-          gid = parts[x].split("=")[1];
+          return parts[x].split("=")[1];
```
Comment (pennello) [inline lib/user.js:98]: Alternatively, you could `break` after assigning `gid`.  Six of one, half a dozen of the other...  Although I guess you'd have to break out of both loops.  `return` is probably better.

File: lib/user.js (hunk 1 above)
Comment (pennello) [inline lib/user.js]: Use `var x`; otherwise it's in the global scope.

**Review by pennello — CHANGES_REQUESTED** [review]

File: lib/user.js [hunk 3]
```diff
@@ -92,7 +92,8 @@ User.prototype.opentable_gid = function(){
   for (i = 0; i < len; i++) {
     if (all_cookie_keys[i].match(cookie_key)) {
       var parts = all_cookie_keys[i].split("&");
-      for (x = 0; x < parts.length; x++) {
+      var len = parts.length;
+      for (var x = 0; x < len; x++) {
```
Comment (pennello) [inline lib/user.js:96]: Cool.

**Review by pennello — APPROVED** [review]

**Review by rajbits — APPROVED** [review]
Included Chris Pennello's suggestion.

**Pull request merged**